tests/specs/api.e2e.js

```diff
@@ -99,4 +102,205 @@ describe('newrelic api', () => {
+      ])
+
+      const pageActions = insResult.request.body.ins
+      expect(pageActions).toBeDefined()
+      expect(pageActions.length).toEqual(1) // exactly 1 PageAction was submitted
+      expect(pageActions[0].actionName).toEqual('finished') // PageAction has actionName = finished
+    })
+  })
+
+  describe('release()', () => {
+    const releaseApiConfig = {
+      init: {
+        page_view_timing: {
+          enabled: false
+        },
+        metrics: {
+          enabled: false
+        }
+      }
+    }
+
+    withBrowsersMatching(reliableUnload)('adds releases to jserrors', async () => {
+      await Promise.all([
+        browser.testHandle.expectRum(),
+        browser.url(await browser.testHandle.assetURL('api/release.html', releaseApiConfig))
+          .then(() => browser.waitForAgentLoad())
+      ])
+
+      const [errorsResult] = await Promise.all([
+        browser.testHandle.expectErrors(),
+        browser.url(await browser.testHandle.assetURL('/'))
+      ])
+
+      expect(errorsResult.request.query.ri).toEqual('{"example":"123","other":"456"}')
+    })
+
+    withBrowsersMatching(reliableUnload)('limits releases to jserrors', async () => {
+      await Promise.all([
+        browser.testHandle.expectRum(),
+        browser.url(await browser.testHandle.assetURL('api/release-too-many.html', releaseApiConfig))
+          .then(() => browser.waitForAgentLoad())
+      ])
+
+      const [errorsResult] = await Promise.all([
+        browser.testHandle.expectErrors(),
+        browser.url(await browser.testHandle.assetURL('/'))
+      ])
+
+      expect(JSON.parse(errorsResult.request.query.ri)).toEqual({
+        one: '1',
+        two: '2',
+        three: '3',
+        four: '4',
+        five: '5',
+        six: '6',
+        seven: '7',
+        eight: '8',
+        nine: '9',
+        ten: '10'
+      })
+    })
+
+    withBrowsersMatching(reliableUnload)('limits size in jserrors payload', async () => {
+      await Promise.all([
+        browser.testHandle.expectRum(),
+        browser.url(await browser.testHandle.assetURL('api/release-too-long.html', releaseApiConfig))
+          .then(() => browser.waitForAgentLoad())
+      ])
+
+      const [errorsResult] = await Promise.all([
+        browser.testHandle.expectErrors(),
+        browser.url(await browser.testHandle.assetURL('/'))
+      ])
+
+      const ninetyNineY = 'yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy'
+      const oneHundredX = 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
+      const twoHundredCharacterString = ninetyNineY + oneHundredX + 'q'
+      const ri = {
+        one: '201',
+        three: twoHundredCharacterString
+      }
+      ri[twoHundredCharacterString] = '2'
+      const queryRi = JSON.parse(errorsResult.request.query.ri)
+      expect(twoHundredCharacterString.length).toBe(200)
+      expect(queryRi).toEqual(ri)
```

Comment:
This test is a bit confusing. First, line 205 is unnecessary. It is testing a variable constructed in the test. Second, could use [`expect.toMatch`](https://jestjs.io/docs/expect#tomatchregexp--string) instead of constructing the string. That would remove lines 196-203 ish.

```suggestion
      const queryRi = JSON.parse(errorsResult.request.query.ri)
      expect(queryRi.one).toEqual('201')
      expect(queryRi.three).toMatch(/y{99}x{100}q/)
      expect(Object.keys(queryRi)).toContain(expect.stringMatching(/y{99}x{100}q/))
```

Reply:
Elegant!